credentials/apps/badges/models.py

```diff
@@ -83,6 +83,29 @@ def save(self, *args, **kwargs):
     @classmethod
     def by_uuid(cls, template_uuid):
         return cls.objects.filter(uuid=template_uuid, origin=cls.ORIGIN).first()
+    
+    def user_progress(username: str) -> float:
+        """
+        Calculate user progress for badge template.
+        """
+        requirements_count = BadgeRequirement.filter(template=self).count()
```

Comment:
`BadgeRequirement.filter >> BadgeRequirement.objects.filter`?
I don't remember we've implemented custom managers...

Test it 😈